oak_functions/proto/unary_server.proto

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+package oak.session.unary.v1;
+
+option java_multiple_files = true;
+option java_package = "oak.session.unary.v1";
+
+message UnaryRequest {
+  bytes body = 1;
+  // Links a sequence of messages, declaring them as part of a single a session.
+  fixed64 session_id = 2;
+}
+
+message UnaryResponse {
+  bytes body = 1;
+}
+
+// Service definition for unary communication with the runtime.
+// It accomplishes the same goal as StreamingSession, just with a differnet
+// request model. Oak Applications can choose either request model in the
+// configuration of the runtime. Eventually we plan to transition all clients to
+// use UnarySessions.
+service UnarySession {
+  // Used to send a sequence of a messages for session key negotiation and
+  // encrypted payload exchange.
+  //
+  // The expected message seqeuence looks as follows:
+  // - Client->Server: `UnaryRequest` with a serialized `ClientHello` message.
+  // - Server->Client: `UnaryResponse` with a serialized `ServerIdentity` message.
+  // - Client->Server: `UnaryRequest` with a serialized `ClientIdentity` message.
+  // - Client->Server: `UnaryRequest` with a serialized `EncryptedData` message.
+  // - Server->Client: `UnaryResponse` with a serialized `EncryptedData` message.
```

Comment:
Perhaps also clarify that once the handshake is complete there can be multiple UnaryRequest/UnaryResponse pairs with encrypted data.